core/src/main/scala/cats/evidence/Is.scala

```diff
@@ -0,0 +1,100 @@
+package cats.evidence
+
+import cats.Id
+
+/**
+ * A value of `A Is B` is proof that the types `A` and `B` are the same. More
+ * powerfully, it asserts that they have the same meaning in all type
+ * contexts. This can be a more powerful assertion than `A =:= B` and is more
+ * easily used in manipulation of types while avoiding (potentially
+ * erroneous) coercions.
+ *
+ * `A Is B` is also known as Leibniz equality.
+ */
+abstract class Is[A, B] extends Serializable {
```

Comment:
I think `Is` is a better name as a replacement of `<:<`.

Did you think about the other names, like `Same`, for this type class?